llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/ContainerRunnerImpl.java

```diff
@@ -488,6 +489,7 @@ public QueryCompleteResponseProto queryComplete(
           fragmentInfo.getFragmentIdentifierString());
         executorService.killFragment(fragmentInfo.getFragmentIdentifierString());
       }
+      fsUgiFactory.closeFileSystemsForQuery(queryIdentifier.toShortString());
```

Comment:
`amReporter.queryComplete(queryIdentifier)` has a null check for queryIdentifier, so can we get a NullPointer here?

Reply:
not anymore,because I changed this to QueryIdentifier according to your other suggestion :D